tools/vscode/ddlog-language/src/good/type.json5

```diff
@@ -0,0 +1,576 @@
+{
+  "repository": {
+        "type": {
+            "patterns": [
+                {
+                    "match": "(\\s*)(?!.*(Some|None|Ok|Err))([A-Z][a-zA-Z0-9_]*)(\\s*)(?=((\\s*)\\{))",
+                    "name": "entity.name.constructor.ddlog"
+                },                
+                {
+                    "match": "('[A-Z][a-zA-Z0-9_]*)",
+                    "name": "entity.name.type-variable.ddlog"
+                }, 
+                {
+                    "name": "entity.name.typedef-typespec.ddlog",                    
+                    "match": "\\b(typedef)\\s*([a-zA-Z_][a-zA-Z0-9_]*)\\s*(=)\\s*(bigint|bit|bool|signed|double|float|string|Vec|Ref|Set|Map)(\\b)",
```

Comment:
these rules for typedef are too specific. The grammar of typedef is 
```
"typedef" type_name (* unique type name *)
            ["<" typevar_name [("," typevar_name)*] ">"] 
            "=" type_spec (* type definition *)
```

I suspect you can parallel this definition in your grammar. Having all these base types is too specific, you should probably have a rule for type_spec.